File: leetcode/easy/divide-array-into-equal-pairs.py
# https://leetcode.com/problems/divide-array-into-equal-pairs/

class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        # seen = set()
        # for n in nums:
        #     if n in seen:
        #         seen.discard(n)
        #     else:
        #         seen.add(n)
        # return not seen

        return all(v % 2 == 0 for v in Counter(nums).values())

File: leetcode/easy/divide-array-into-equal-pairs.ts
// https://leetcode.com/problems/divide-array-into-equal-pairs/

// MINE: map
/*
function divideArray(nums: number[]): boolean {
    const map = new Map()
    for(let n of nums) map.set(n, map.get(n)+1 || 1)
    for(let [key, val] of map) if(val%2 !== 0) return false
    return true
};
*/

// ANSWER
function divideArray(nums: number[]): boolean {
  const set = new Set()
  for (let n of nums) set.has(n) ? set.delete(n) : set.add(n)
  return !set.size
}
